### PR TITLE
Add xip patch chart

### PR DESCRIPTION
**Pull request by kfurgol**
**Description**

Changes proposed in this pull request:

- Added xip patch chart
- Added xip-installation.yaml (please suggest better place for that file)
- xip-patch component - please check #2055

**Related issue(s)**
Adopt kyma installation for xip.io domain #1871

### CODE REVIEW

File: installation/resources/xip-installation.yaml [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  name: service-catalog-overrides
+  namespace: kyma-installer
+  labels:
+    installer: overrides
+    component: service-catalog
+data:
+  etcd-stateful.etcd.resources.limits.memory: 512Mi
+
+---
+apiVersion: "installer.kyma-project.io/v1alpha1"
```
Comment (sjanota) [inline installation/resources/xip-installation.yaml]: Why you copied whole installer.yaml, not only installation-cr.yaml?

File: resources/xip-patch/templates/clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: xip-patch
+rules:
+- apiGroups: ["*"]
```
Comment (sjanota) [inline resources/xip-patch/templates/clusterrole.yaml]: Specify api groups here. With `*` this script would have access to all resources named `configmaps` and `scripts`, regardless of their apiGroup. In Knative for exmaple is resource `service` and we don't want the script to have access to it.

File: resources/xip-patch/templates/clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  resources: ["configmaps", "services"]
+  verbs: ["*"]
```
Comment (sjanota) [inline resources/xip-patch/templates/clusterrole.yaml]: You don't need write access to `service`. Split it into two rules: one with read-write access to `configmaps` and second with read-only access to `services`.

**Comment by sjanota** [comment]
I've tested it on my own GKE cluster. Everything works fine.

File: resources/xip-patch/templates/clusterrole.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: xip-patch
```
Comment (Demonsthere) [inline resources/xip-patch/templates/clusterrole.yaml]: The name element in the chart should contain the Release.name too: `{{ .Release.Name }}-xip-patch`

File: resources/xip-patch/templates/job.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+spec:
+  template:
+    metadata:
+      name: xip-patch
+    spec:
+      serviceAccountName: xip-patch
+      restartPolicy: OnFailure
+      containers:
+      - name: xip-patch
+        image: eu.gcr.io/kyma-project/pr/xip-patch:PR-2055
```
Comment (Demonsthere) [inline resources/xip-patch/templates/job.yaml]: I see you have created default values, why are you not using them here?

File: resources/xip-patch/values.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+global:
```
Comment (Demonsthere) [inline resources/xip-patch/values.yaml]: I would avoid the `global` keyword here, just keep the structure:
```yaml
---
containerRegistry:
  path: eu.gcr.io/kyma-project
xip_patch:
  dir: pr/xip-patch/
  version: PR-2055
```

File: resources/xip-patch/templates/clusterrole.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
Comment (sjanota) [inline resources/xip-patch/templates/clusterrole.yaml]: When thing about it it probably shouldn't be `ClusterRole`, but two separate `Role`s. One of them should allow access to services in `istio-system` and second to config maps in `kyma-installer`. Current scope is too broad.

**Comment by sjanota** [comment]
@jakkab I don't agree. As far as I'm aware the helm convention is to have file per resource (at least that's what we do in Kyma). I don't claim it's wise convention, though ;). I'd say let's stay with file per resource

**Pull request merged**